application-tour-ui/src/main/resources/TourCode/StepsListing.xml

```diff
@@ -101,7 +100,9 @@
                      $!placement
                   &lt;/td&gt;
                   &lt;td&gt;
-                     $!targetPage
+                     $!targetPage &lt;br&gt;
+                     $!action &lt;br&gt;
+                     $!queryString
```

Comment:
The page can also be left empty (to take the previous value by default), so in that case you would have something like `view/?query` which is ugly in my opinion.